ocs_ci/utility/storage_cluster_setup.py

```diff
@@ -0,0 +1,650 @@
+from ocs_ci.deployment.helpers import storage_class
+import tempfile
+import yaml
+from copy import deepcopy
+from ocs_ci.utility import (
+    templating,
+    kms as KMS,
+    pgsql,
+    version,
+)
+from ocs_ci.framework import config, merge_dict
+import logging
+from ocs_ci.deployment.helpers.mcg_helpers import (
+    mcg_only_deployment,
+)
+from ocs_ci.ocs.node import (
+    get_node_objs,
+)
+from ocs_ci.ocs.resources.storage_cluster import (
+    setup_ceph_debug,
+)
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.helpers import helpers
+from ocs_ci.framework.logger_helper import log_step
+from ocs_ci.deployment.encryption import add_in_transit_encryption_to_cluster_data
+from ocs_ci.ocs import constants, ocp, defaults
+from ocs_ci.ocs.exceptions import (
+    UnavailableResourceException,
+    UnsupportedFeatureError,
+)
+from ocs_ci.utility.utils import (
+    get_az_count,
+    run_cmd,
+)
+
+logger = logging.getLogger(__name__)
+
+
+# create custom storage class for StorageCluster CR if necessary
+class StorageClusterSetup(object):
+    def __init__(self):
+        self.custom_storage_class_path = None
+        self.namespace = constants.OPENSHIFT_STORAGE_NAMESPACE
+        self.platform = config.ENV_DATA["platform"]
+        self.storage_class = constants.LOCALSTORAGE_SC
```

Comment:
Should we default to retrieving the default storageclass the same way we do for ODF deployments? Then we can override this value in the event we are deploying with local storage, or update `get_storageclass()` to return the proper value if we are configured for local storage.

```suggestion
        self.storage_class = storage_class.get_storageclass()
```